packages/js/e2e-core-tests/specs/merchant/wp-admin-page-loads.test.js

```diff
@@ -0,0 +1,131 @@
+/**
+ * Internal dependencies
+ */
+const { merchant } = require( '@woocommerce/e2e-utils' );
+
+/**
+ * External dependencies
+ */
+const { it, describe, beforeAll } = require( '@jest/globals' );
+
+// Top-level menu items and their associated sub-menus
+const menus = [
+	[
+		'WooCommerce',
+		'#adminmenu > li:nth-child(8) > a',
+		[
+			[
+				'Home',
+				'#toplevel_page_woocommerce > ul > li:nth-child(2)',
+				'Home',
+			],
+			[
+				'Orders',
+				'#toplevel_page_woocommerce > ul > li:nth-child(3)',
+				'Orders',
+			],
+			[
+				'Reports',
+				'#toplevel_page_woocommerce > ul > li:nth-child(6)',
+				'Orders',
+			],
+			[
+				'Settings',
+				'#toplevel_page_woocommerce > ul > li:nth-child(7)',
+				'General',
+			],
+			[
+				'Status',
+				'#toplevel_page_woocommerce > ul > li:nth-child(8)',
+				'System status',
+			],
+			// [ 'Extensions', '#toplevel_page_woocommerce > ul > li:nth-child(9)', 'Extensions' ],
+		],
+	],
+	[
+		'Products',
+		'#adminmenu > li:nth-child(9) > a',
+		[
+			[
+				'All Products',
+				'#menu-posts-product > ul > li:nth-child(2)',
+				'Products',
+			],
+			[
+				'Add New',
+				'#menu-posts-product > ul > li:nth-child(3)',
+				'Add New',
+			],
+			[
+				'Categories',
+				'#menu-posts-product > ul > li:nth-child(4)',
+				'Product categories',
+			],
+			[
+				'Product tags',
+				'#menu-posts-product > ul > li:nth-child(5)',
+				'Product tags',
+			],
+			[
+				'Attributes',
+				'#menu-posts-product > ul > li:nth-child(6)',
+				'Attributes',
+			],
+		],
+	],
+	[
+		'Marketing',
+		'#adminmenu > li:nth-child(11) > a',
+		[
+			[
+				'Overview',
+				'#toplevel_page_woocommerce-marketing > ul > li:nth-child(2)',
+				'Overview',
+			],
+			[
+				'Coupons',
+				'#toplevel_page_woocommerce-marketing > ul > li:nth-child(3)',
+				'Coupons',
+			],
+		],
+	],
+];
+
+const runPageLoadTest = () => {
+	describe.each( menus )(
+		' %s > Opening Top Level Pages',
+		( menuTitle, menuElement, subMenus ) => {
+			beforeAll( async () => {
+				await merchant.login();
+			} );
+
+			it.each( subMenus )(
+				'can see %s page properly',
+				async ( subMenuTitle, subMenuElement, subMenuText ) => {
+					// Go to Top Level Menu
+					await Promise.all( [
+						page.click( menuElement ),
+						page.waitForNavigation( { waitUntil: 'networkidle0' } ),
+						page.setViewport( {
```

Comment:
Can you check to see if this will work in the `beforeAll()`

Reply:
The viewport is now being set in the `beforeAll` block, but now, page load tests are failing because the side menu isn't expanded. My assumption is that the viewport is being resized in between individual test steps.